packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -242,24 +242,34 @@ export class SummarizerNode implements IRootSummarizerNode {
         );
         return { latestSummaryUpdated: true, wasSummaryTracked: false, snapshot: snapshotTree };
     }
-
+/**
+ * Called when we get an ack from the server for a summary we've just sent. Updates the reference state of this node
+ * from the state in the pending summary queue.
+ * @param proposalHandle - Handle for the current proposal.
+ * @param referenceSequenceNumber -  reference sequence number of sent summary.
+ */
     protected refreshLatestSummaryFromPending(
         proposalHandle: string,
         referenceSequenceNumber: number,
     ): void {
         const summaryNode = this.pendingSummaries.get(proposalHandle);
         if (summaryNode === undefined) {
-            // This should only happen if parent skipped recursion AND no prior summary existed.
-            assert(
-                this._latestSummary === undefined,
-                0x1a6 /* "Not found pending summary, but this node has previously completed a summary" */,
-            );
-            return;
-        } else {
-            assert(
-                referenceSequenceNumber === summaryNode.referenceSequenceNumber,
-                0x1a7 /* Pending summary reference sequence number should be consistent */,
-            );
+            if (summaryNode === undefined) {
+                // This should only happen if parent skipped recursion AND no prior summary existed.
+                assert(
+                    this._latestSummary === undefined,
+                    0x1a6 /* "Not found pending summary, but this node has previously completed a summary" */,
+                );
+                return;
+            } else {
+                assert(
+                    referenceSequenceNumber === summaryNode.referenceSequenceNumber,
+                    0x1a7 /* Pending summary reference sequence number should be consistent */,
+                );
+
+                // Clear earlier pending summaries
+                this.pendingSummaries.delete(proposalHandle);
+            }
```

Comment:
Looks like `summaryNode === undefined is done` twice this shouldn't be green right?